internal/controller/gcpbuild_controller.go

```diff
@@ -178,19 +180,25 @@ func (r *GCPBuildReconciler) reconcileNormal(ctx context.Context, buildScope *sc
 		subnets.New(buildScope),
 		// TODO, make sure to provide a way of connecting to this machine.
 		instances.New(buildScope),
+		images.New(buildScope),
 		// TODO, createConnectionSecret.
 		// TODO exportImage,
 		// TODO cleanup GCP resources.
 	}
-
-	for _, r := range reconcilers {
-		if err := r.Reconcile(ctx); err != nil {
-			logger.Error(err, "Reconcile error")
-			record.Warnf(buildScope.GCPBuild, "GCPBuildReconciler", "Reconcile error - %v", err)
-			return ctrl.Result{}, err
+	if !buildScope.IsReady() {
+		for _, r := range reconcilers {
+			if err := r.Reconcile(ctx); err != nil {
+				logger.Error(err, "Reconcile error")
+				record.Warnf(buildScope.GCPBuild, "GCPBuildReconciler", "Reconcile error - %v", err)
+				return ctrl.Result{}, err
+			}
 		}
 	}
 
+	if buildScope.IsReady() && !buildScope.IsCleanedUp() {
```

Comment:
Another point, it's not ideal to call `reconcileDelete` cause it will.
remove the finalizer.
create a new function `cleanupResources()` and call it here and in `reconcileDelete`

Reply:
Why do we need the infrastructure finalizer if the infrastructure is cleaned up?

Reply:
> You need to add a comment explaining why you need to execute reconcileDelete.

When the Image is ready and the Infrastructure is not cleaned up yet, we need to call reconcileDelete to delete the infrastructure and the finalizer of the infrastructure.